### PR TITLE
Switch from `failure` to `anyhow`

### PR DESCRIPTION
**Pull request by adumbidiot**
This PR switches the error  type of this CLI from `failure` to `anyhow` since `failure` is [deprecated](https://github.com/rust-lang-deprecated/failure) now.

### CODE REVIEW

**Comment by adumbidiot** [comment]
CI failure is due to https://github.com/rust-lang/cargo/issues/9788 I believe

File: src/main.rs [hunk 1]
```diff
@@ -218,7 +218,7 @@ fn main() {
             debug!("Writing timestamp file in: {:?}", path);
             match Timestamp::new().store(path.as_path()) {
                 Ok(_) => {}
-                Err(e) => error!("Failed to write timestamp file: {}", e),
+                Err(e) => error!("Failed to write timestamp file: {:?}", e),
```
Comment (jyn514) [inline src/main.rs:221]: why did this change?

**Review by jyn514 — COMMENTED** [review]
:+1: awesome to see this cuts down on the number of dependencies! If you rebase the PR so CI reruns I'm happy to merge it :)

File: src/main.rs (hunk 1 above)
Comment (adumbidiot) [reply src/main.rs:221]: anyhow has, to my knowledge, 2 different error output options. One just prints a simple one-line error message, which is usually just the error context. The other is more detailed and prints the entire error chain. Here is an example of the second option, taken from anyhow's docs: 
```
Error: Failed to read instrs from ./path/to/instrs.json

Caused by:
    No such file or directory (os error 2)
```

I opted for the second option as it is more detailed and more helpful if one runs into an error.

Comment (jyn514) [reply src/main.rs:221]: ah perfect - for some reason I thought it was `{:#}` but the docs say it's `{:?}` like you have here: https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations

**Pull request merged**